Code/Editor/Plugins/ProjectSettingsTool/ProjectSettingsContainer.cpp

```diff
@@ -155,80 +166,104 @@ namespace ProjectSettingsTool
-        LoadProjectJsonData();
+        LoadJson(m_projectJson);
     }
 
-    void ProjectSettingsContainer::SavePlistsData()
+    void ProjectSettingsContainer::SaveAllPlatformsData()
     {
-        for (AZStd::pair<PlatformId, PlistSettings>& plist : m_pListsMap)
+        for (const auto& platformData : m_platformSettingsMap)
         {
-            LoadPlist(plist.second);
+            const PlatformSettings& platformSettings = platformData.second;
+
+            if (AZStd::holds_alternative<JsonSettings>(platformSettings))
+            {
+                SaveJson(AZStd::get<JsonSettings>(platformSettings));
+            }
+            else
+            {
+                SavePlist(AZStd::get<PlistSettings>(platformSettings));
+            }
         }
     }
 
-    void ProjectSettingsContainer::SavePlistData(const Platform& plat)
+    void ProjectSettingsContainer::SavePlatformData(const Platform& plat)
     {
-        PlistSettings* settings = GetPlistSettingsForPlatform(plat);
-        if (settings != nullptr)
+        const PlatformSettings* platformSettings = GetPlatformData(plat);
+        if (platformSettings != nullptr)
         {
-            SavePlist(*settings);
+            if (AZStd::holds_alternative<JsonSettings>(*platformSettings))
+            {
+                SaveJson(AZStd::get<JsonSettings>(*platformSettings));
+            }
+            else
+            {
+                SavePlist(AZStd::get<PlistSettings>(*platformSettings));
+            }
         }
     }
 
-    void ProjectSettingsContainer::ReloadPlistData()
+    void ProjectSettingsContainer::ReloadAllPlatformsData()
     {
-        for (AZStd::pair<PlatformId, PlistSettings>& plist : m_pListsMap)
+        for (AZStd::pair<PlatformId, PlatformSettings>& platformData : m_platformSettingsMap)
         {
-            PlistSettings& plistSettings = plist.second;
+            PlatformSettings& platformSettings = platformData.second;
 
-            plistSettings.m_document.reset(new XmlDocument());
-            LoadPlist(plistSettings);
+            if (AZStd::holds_alternative<JsonSettings>(platformSettings))
+            {
+                auto& jsonSettings = AZStd::get<JsonSettings>(platformSettings);
+                jsonSettings.m_document.reset(new rapidjson::Document);
+                LoadJson(jsonSettings);
+            }
+            else
+            {
+                auto& plistSettings = AZStd::get<PlistSettings>(platformSettings);
+                plistSettings.m_document.reset(new XmlDocument());
+                LoadPlist(plistSettings);
+            }
         }
     }
 
     rapidjson::Document& ProjectSettingsContainer::GetProjectJsonDocument()
     {
-        return *m_projectJson.m_document.get();
+        return *m_projectJson.m_document;
     }
 
     AZ::Outcome<rapidjson::Value*, void> ProjectSettingsContainer::GetProjectJsonValue(const char* key)
+    {
+        return GetJsonValue(*m_projectJson.m_document, key);
```

Comment:
nit: I would just call `GetProjectJsonDocument()` here.

Reply:
It's not getting the document, it's getting a value inside the document, ergo called `GetProjectJsonValue`.